calico/felix/frules.py

```diff
@@ -409,7 +409,9 @@ def interface_to_chain_suffix(config, iface_name):
     :param iface_name: The interface name
     :returns string: the suffix (shortened if necessary)
     """
-    suffix = iface_name.replace(config.IFACE_PREFIX, "", 1)
-    # The suffix is surely not very long, but make sure.
-    suffix = futils.uniquely_shorten(suffix, 16)
-    return suffix
+    for prefix in config.IFACE_PREFIX:
+        if iface_name.startswith(prefix):
+            iface_name = iface_name[len(prefix):]
+            break
```

Comment:
D'oh, you're right, sorry about that.

Yes, I think your idea about `interface_to_chain_suffix` is good, and means that we don't need the validation in `_validate_cfg`.